TST: Failures due to astropy unit string reorder

Failures in dev job caused by:

* https://github.com/astropy/astropy/pull/14439

```
=================================== FAILURES ===================================
_______________________ test_fits_image_hdu_with_microns _______________________

...
>       assert label_mouseover.as_text() == ('Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',  # noqa
                                             'World 13h41m45.5759s +27d00m12.3044s (ICRS)',
                                             '205.4398995981 27.0034178810 (deg)')  # noqa
E       AssertionError: assert ('Pixel x=00....178810 (deg)') == ('Pixel x=00....178810 (deg)')
E         At index 0 diff: 'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom s cm2)' != 'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)'
E         Full diff:
E           (
E         -  'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',
E         ?                                                                   --
E         +  'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom s cm2)',
E         ?                                                                ++
E            'World 13h41m45.5759s +27d00m12.3044s (ICRS)',
E            '205.4398995981 27.0034178810 (deg)',
E           )

jdaviz/configs/cubeviz/plugins/tests/test_parsers.py:48: AssertionError
_____________________ test_fits_image_hdu_parse_from_file ______________________

...
>       assert label_mouseover.as_text() == ('Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',  # noqa
                                             'World 13h41m46.5994s +26d59m58.6136s (ICRS)',
                                             '205.4441642302 26.9996148973 (deg)')
E       AssertionError: assert ('Pixel x=00....148973 (deg)') == ('Pixel x=00....148973 (deg)')
E         At index 0 diff: 'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom s cm2)' != 'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)'
E         Full diff:
E           (
E         -  'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',
E         ?                                                                   --
E         +  'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom s cm2)',
E         ?                                                                ++
E            'World 13h41m46.5994s +26d59m58.6136s (ICRS)',
E            '205.4441642302 26.9996148973 (deg)',
E           )

jdaviz/configs/cubeviz/plugins/tests/test_parsers.py:100: AssertionError
____________________________ test_spectrum3d_parse _____________________________

...
>       assert label_mouseover.as_text() == ('Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',  # noqa
                                             'World 13h41m46.5994s +26d59m58.6136s (ICRS)',
                                             '205.4441642302 26.9996148973 (deg)')
E       AssertionError: assert ('Pixel x=00....148973 (deg)') == ('Pixel x=00....148973 (deg)')
E         At index 0 diff: 'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom s cm2)' != 'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)'
E         Full diff:
E           (
E         -  'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',
E         ?                                                                   --
E         +  'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom s cm2)',
E         ?                                                                ++
E            'World 13h41m46.5994s +26d59m58.6136s (ICRS)',
E            '205.4441642302 26.9996148973 (deg)',
E           )

jdaviz/configs/cubeviz/plugins/tests/test_parsers.py:129: AssertionError
______________ [doctest] jdaviz.models.physical_models.BlackBody _______________
057         and re-added internally when returning results.  Note that this can result in
058         ambiguous and unexpected results.  To avoid confusion, pass unitless
059         `scale` and `output_units` separately.
060 
061     Examples
062     --------
063     >>> from astropy.modeling import models
064     >>> from astropy import units as u
065     >>> bb = models.BlackBody(temperature=5000*u.K)
066     >>> bb(6000 * u.AA)  # doctest: +FLOAT_CMP
Expected:
    <Quantity 1.53254685e-05 erg / (cm2 Hz s sr)>
Got:
    <Quantity 1.53254685e-05 erg / (Hz s sr cm2)>

jdaviz/models/physical_models.py:66: DocTestFailure
...
============= 4 failed, 408 passed, 2 skipped in 265.68s (0:04:25) =============
```

[🐱](https://jira.stsci.edu/browse/JDAT-3165)

--- a/jdaviz/configs/cubeviz/plugins/tests/test_parsers.py
+++ b/jdaviz/configs/cubeviz/plugins/tests/test_parsers.py
@@ -1,6 +1,8 @@
+import astropy
 import numpy as np
 import pytest
 from astropy import units as u
+from astropy.utils.introspection import minversion
 from astropy.wcs import WCS
 from specutils import Spectrum1D
 from glue.core.roi import XRangeROI
@@ -8,6 +10,8 @@ from glue_astronomy.translators.spectrum1d import PaddedSpectrumWCS
 from numpy.testing import assert_allclose, assert_array_equal
 
 from jdaviz.utils import PRIHDR_KEY
+
+ASTROPY_LT_5_3 = not minversion(astropy, "5.3.dev")
 
 
 @pytest.mark.filterwarnings('ignore')
@@ -45,7 +49,11 @@ def test_fits_image_hdu_with_microns(image_cube_hdu_obj_microns, cubeviz_helper)
     label_mouseover = cubeviz_helper.app.session.application._tools['g-coords-info']
     label_mouseover._viewer_mouse_event(flux_viewer,
                                         {'event': 'mousemove', 'domain': {'x': 0, 'y': 0}})
-    assert label_mouseover.as_text() == ('Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',  # noqa
+    if ASTROPY_LT_5_3:
+        flux_unit_str = "erg / (Angstrom cm2 s)"
+    else:
+        flux_unit_str = "erg / (Angstrom s cm2)"
+    assert label_mouseover.as_text() == (f'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 {flux_unit_str}',  # noqa
                                          'World 13h41m45.5759s +27d00m12.3044s (ICRS)',
                                          '205.4398995981 27.0034178810 (deg)')  # noqa
     unc_viewer = cubeviz_helper.app.get_viewer('uncert-viewer')
@@ -97,7 +105,11 @@ def test_fits_image_hdu_parse_from_file(tmpdir, image_cube_hdu_obj, cubeviz_help
     label_mouseover = cubeviz_helper.app.session.application._tools['g-coords-info']
     label_mouseover._viewer_mouse_event(flux_viewer,
                                         {'event': 'mousemove', 'domain': {'x': 0, 'y': 0}})
-    assert label_mouseover.as_text() == ('Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',  # noqa
+    if ASTROPY_LT_5_3:
+        flux_unit_str = "erg / (Angstrom cm2 s)"
+    else:
+        flux_unit_str = "erg / (Angstrom s cm2)"
+    assert label_mouseover.as_text() == (f'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 {flux_unit_str}',  # noqa
                                          'World 13h41m46.5994s +26d59m58.6136s (ICRS)',
                                          '205.4441642302 26.9996148973 (deg)')
 
@@ -126,7 +138,11 @@ def test_spectrum3d_parse(image_cube_hdu_obj, cubeviz_helper):
     label_mouseover = cubeviz_helper.app.session.application._tools['g-coords-info']
     label_mouseover._viewer_mouse_event(flux_viewer,
                                         {'event': 'mousemove', 'domain': {'x': 0, 'y': 0}})
-    assert label_mouseover.as_text() == ('Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',  # noqa
+    if ASTROPY_LT_5_3:
+        flux_unit_str = "erg / (Angstrom cm2 s)"
+    else:
+        flux_unit_str = "erg / (Angstrom s cm2)"
+    assert label_mouseover.as_text() == (f'Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 {flux_unit_str}',  # noqa
                                          'World 13h41m46.5994s +26d59m58.6136s (ICRS)',
                                          '205.4441642302 26.9996148973 (deg)')
 

--- a/jdaviz/models/physical_models.py
+++ b/jdaviz/models/physical_models.py
@@ -17,6 +17,9 @@ from astropy.modeling.parameters import Parameter
 
 __all__ = ["BlackBody"]
 
+# ASTROPY_LT_5_3
+__doctest_requires__ = {"BlackBody": ["astropy<5.3"]}
+
 
 class BlackBody(Fittable1DModel):
     """

